File: CODE-RUN/server/main/app/routers/tag.py
# 표준 라이브러리
from os import path
from sys import path as pth
from typing import Optional

# 서드파티 라이브러리
from fastapi import APIRouter, Depends, HTTPException, Header
from sqlalchemy.orm import Session

# 로컬 라이브러리
pth.append(path.dirname(path.abspath(path.dirname(__file__))))
from database import models, schemas
from dependency import get_db
from routers.user import get_current_user


router = APIRouter()


@router.get("/api/tag/language", tags=["tag"], description="언어 태그 리스트 확인")
def get_language_tag(db: Session = Depends(get_db)):
    return {"data": db.query(models.LanguageTag).all()}


@router.get("/api/tag/algorithm", tags=["tag"], description="알고리즘 태그 리스트 확인")
def get_algorithm_tag(db: Session = Depends(get_db)):
    return {"data": db.query(models.AlgorithmTag).all()}


@router.get("/api/tag/subject", tags=["tag"], description="CS 태그 리스트 확인")
def get_subject_tag(db: Session = Depends(get_db)):
    return {"data": db.query(models.SubjectTag).all()}

File: CODE-RUN/client/src/_api/Tag.js
import axios from 'axios';
import { SERVER } from 'Config.js';

//언어들 목록
export function fetchLanguageTag() {
    return axios.get(`${SERVER}/api/tag/language`);
}
// 알고리즘 개념 목록
export function fetchAlgorithmTag() {
    return axios.get(`${SERVER}/api/tag/algorithm`);
}
// CS 과목 목록
export function fetchCSTag() {
    return axios.get(`${SERVER}/api/tag/subject`);
}
